### PR TITLE
AWS RDS support

### PR DESCRIPTION
**Pull request by karikarshivani**
- [x] Bump version
- [x] Run InSpec profile against local and RDS instances
- [x] Change `concat` to `+`
- [x] Update the README to show AWS RDS support
- [ ] Update branch name on the overlay after testing is complete
- [ ] Merge both the PRs
- [ ] Remove redundant repos (AWS RDS baseline + CMS AWS RDS overlay)

### CODE REVIEW

**Comment by ejaronne** [comment]
add rdsadmin to mysql_administrative_users input in inspec.yml

SV-235096: exclude users in mysql_administrative_users input from second describe block evaluation

**Comment by ejaronne** [comment]
SV-235104: "match_array" isn't good. The describe block should check that any grantees are in the array, not an exact match.

**Comment by ejaronne** [comment]
SV-235135: If AWS RDS is true, Set to "Not applicable since these requirements are handled within AWS RDS"

**Comment by karikarshivani** [comment]
> SV-235104: "match_array" isn't good. The describe block should check that any grantees are in the array, not an exact match.

I was thinking the same thing - changed the matcher to `be_in`! I just pushed a change that changes the output from:
Screen Shot 2023-03-21 at 12 42 04 PM
To:
Screen Shot 2023-03-21 at 12 42 22 PM

With rationale: authorized users should be known, but expected users not authorized in the DB is acceptable.

**Comment by karikarshivani** [comment]
> SV-235138: Add rdsadmin to authorized_password_users

This may need some discussion before I finalize a change. When I try to retrieve accounts that use passwords, I get this on the localhost:
Screen Shot 2023-03-21 at 1 13 36 PM
And this on the RDS:
Screen Shot 2023-03-21 at 1 13 59 PM

The [actual query in the control](https://github.com/mitre/oracle-mysql-8-stig-baseline/blob/main/controls/SV-235138.rb#L83:L89) has an additional condition - `AND NOT (user like 'mysql.%' or user ='root')` - which leaves us with nothing on the hosted but with **admin** and **rdsadmin** on the RDS instance. Which value(s) should be added to the `authorized_password_users` input?

**Comment by karikarshivani** [comment]
SV-235150 mysql_administrative_users additional values?

Hosted:
Screen Shot 2023-03-21 at 1 34 46 PM


RDS:
Screen Shot 2023-03-21 at 1 34 29 PM

**Comment by ejaronne** [comment]
> SV-235150 mysql_administrative_users additional values?
> 
> Hosted: Screen Shot 2023-03-21 at 1 34 46 PM
> 
> RDS: Screen Shot 2023-03-21 at 1 34 29 PM

ONLY rdsadmin.

NOTE: don't add 'admin' to code/inputs. That is simply the user-defined administrative account when setting up the database which can be anything. rdsadmin is built-in and not a choice.

**Comment by ejaronne** [comment]
SV-235155  exclude all of these from last describe block which cannot have their settings modified anyway in RDS:
mysql/rds_configuration   
mysql/rds_history 
mysql/rds_replication_status 
mysql/rds_global_status_history   
mysql/rds_global_status_history_old 
mysql/rds_heartbeat2              
mysql/rds_sysinfo

**Comment by ejaronne** [comment]
SV-235180 add default rds stored procedures for exclusion

**Comment by karikarshivani** [comment]
> SV-235146: add /tmp/mysql.sock to mysql_sockets

Updated 146 and 167 with the same changes - updating matcher after adding `/tmp/mysql.sock` to the default values.

**Comment by karikarshivani** [comment]
> SV-235155,192 exclude all of these from last describe block which cannot have their settings modified anyway in RDS: mysql/rds_configuration mysql/rds_history mysql/rds_replication_status mysql/rds_global_status_history mysql/rds_global_status_history_old mysql/rds_heartbeat2 mysql/rds_sysinfo

Used an approach for 155, when confirmed, I'll replicate it in 192 and 193

**Comment by karikarshivani** [comment]
SV-235169
I had originally added an `aws_rds` condition to skip over both the describe blocks with the `file` resource in them. I see here that we can access the current permissions on the cnf files:
Screen Shot 2023-03-30 at 12 04 51 PM

But can't `chmod` so I'm leaving the condition as it is:
Screen Shot 2023-03-30 at 12 06 58 PM

**Comment by karikarshivani** [comment]
> SV-235155,192 exclude all of these from last describe block which cannot have their settings modified anyway in RDS: mysql/rds_configuration mysql/rds_history mysql/rds_replication_status mysql/rds_global_status_history mysql/rds_global_status_history_old mysql/rds_heartbeat2 mysql/rds_sysinfo

Implemented for 155, 192, and 193

**Comment by karikarshivani** [comment]
> SV-235180 add default rds stored procedures for exclusion

Implemented for 180 and 181

File: controls/SV-235135.rb [hunk 1]
```diff
@@ -160,55 +160,63 @@
...
+
+    full_key_path = "#{ssl_params.column('@@datadir').join}#{ssl_params.column('@@ssl_key').join}"
+    describe "SSL Private Key file: #{full_key_path}" do
+      subject { file(full_key_path) }
+      its('owner') { should cmp 'mysql' }
+      its('group') { should cmp 'mysql' }
+      it { should_not be_more_permissive_than('0600') }
+    end
+  else
+    impact 0.0
```
Comment (ejaronne) [inline controls/SV-235135.rb:217]: Like SV-235134, since we still do one test for RDS instances, we shouldn't make this NA.

**Review by ejaronne — CHANGES_REQUESTED** [review]
Will approve pending https://github.com/mitre/oracle-mysql-8-stig-baseline/pull/9#discussion_r1154602504 update.

File: controls/SV-235135.rb (hunk 1 above)
Comment (karikarshivani) [reply controls/SV-235135.rb]: Removed the `else` block

**Pull request merged**